app/src/lib/get-main-guid.ts

```diff
@@ -0,0 +1,45 @@
+import { app } from 'electron'
+import { readFile, writeFile } from 'fs/promises'
+import { join } from 'path'
+import { uuid } from './uuid'
+
+let cachedGUID: string | null = null
+
+/** Get the GUID for the Main process. */
+export async function getMainGUID(): Promise<string> {
+  if (!cachedGUID) {
+    cachedGUID = await readGUIDFile()
+  }
+
+  return cachedGUID
+}
+
+/** Reads the persisted GUID from the .guid file or generates a new one */
+export async function readGUIDFile(): Promise<string> {
+  let guid = undefined
+
+  try {
+    guid = await readFile(getGUIDPath(), 'utf8')
+
+    // Validate (at least) the GUID by its length
+    if (guid.length !== 36) {
+      guid = undefined
+    }
+  } catch (e) {}
+
+  if (guid === undefined) {
+    guid = uuid()
+    await saveGUIDFile(guid)
+  }
+
+  return guid
+}
+
+/** Saves the GUID to the .guid file */
+export async function saveGUIDFile(guid: string) {
```

Comment:
I don't think it's necessary, but while writing the explanation about it, I realized the renderer should be getting the GUID from `getMainGUID` (via IPC, of course) instead of using `readGUID` 🤦  that way the renderer will use whatever the main process had cached or will populate its cache at the same time.

In any case… the reason why I think it's not necessary is because `saveGUIDFile` is used:
1. When the main process tries to populate its cache, so the cached value is already set by `getMainGUID`
2. When the renderer process migrates an existing GUID from the LocalStorage. In that case, the main process would only have a GUID if it crashed before the migration. If the main process didn't crash (which is the only way to get to the migration step?), it shouldn't have anything cached and all should be good.

Does that make sense? 😳

Reply:
Actually, maybe it makes sense since I'm caching the migrated GUID on the renderer process even when persistence fails. I'm gonna change logic a bit:
1. Don't remove existing GUID from the LocalStorage if persistence failed.
2. Update cached GUID in main process on migration